test/e2e/aws_eip_for_nlb_test.go

```diff
@@ -0,0 +1,434 @@
+
+	// Now, update the IngressController to use invalid (non-existent) eipAllocations.
+	t.Logf("updating ingresscontroller %q to use invalid eipAlocations", ic.Name)
+	invalidEIPAllocations := createInvalidEIPAllocations(t)
+	fmt.Printf("invalidEIPAllocations %v", invalidEIPAllocations)
+	if err = updateIngressControllerWithRetryOnConflict(t, name, 5*time.Minute, func(ic *operatorv1.IngressController) {
+		ic.Spec.EndpointPublishingStrategy.LoadBalancer.ProviderParameters = &operatorv1.ProviderLoadBalancerParameters{
+			Type: operatorv1.AWSLoadBalancerProvider,
+			AWS: &operatorv1.AWSLoadBalancerParameters{
+				Type: operatorv1.AWSNetworkLoadBalancer,
+				NetworkLoadBalancerParameters: &operatorv1.AWSNetworkLoadBalancerParameters{
+					EIPAllocations: invalidEIPAllocations,
+				},
+			},
+		}
+	}); err != nil {
+		t.Fatalf("failed to update ingresscontroller: %v", err)
+	}
+
+	// Wait for the LoadBalancerProgressing to become True, which represents the message telling
+	// the cluster admin to delete the service to effectuate the eipAllocations.
+	loadBalancerProgressingTrue := operatorv1.OperatorCondition{
+		Type:   ingress.IngressControllerLoadBalancerProgressingConditionType,
+		Status: operatorv1.ConditionTrue,
+	}
+	if err = waitForIngressControllerCondition(t, kclient, 5*time.Minute, name, loadBalancerProgressingTrue); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
+	}
+
+	t.Logf("recreating the service to effectuate the eipAllocations: %s/%s", controller.LoadBalancerServiceName(ic).Namespace, controller.LoadBalancerServiceName(ic).Namespace)
+	if err = recreateIngressControllerService(t, ic); err != nil {
+		t.Fatalf("failed to delete and recreate service: %v", err)
+	}
+
+	// Ensure the service's load-balancer status changes, and verify we get the expected eipAllocation annotation on the service.
+	if err = waitForEIPAllocationAnnotation(t, ic, invalidEIPAllocations); err != nil {
+		t.Fatalf("failed to wait for expected eipAllocation annotation on service: %v", err)
+	}
+
+	// Verify the eipAllocations status field is configured to what we expect.
+	if err = verifyIngressControllerEIPAllocationStatus(t, name); err != nil {
+		t.Fatalf("failed to verify ingresscontroller eipAllocation status: %v", err)
+	}
+
+	// Since the eipAllocations are invalid, the load balancer will fail to provision and set LoadBalancerReady=False.
+	loadBalancerReadyFalse := operatorv1.OperatorCondition{
+		Type:   operatorv1.LoadBalancerReadyIngressConditionType,
+		Status: operatorv1.ConditionFalse,
+	}
+	if err = waitForIngressControllerCondition(t, kclient, 5*time.Minute, name, loadBalancerReadyFalse); err != nil {
+		t.Fatalf("failed to observe expected conditions: %v", err)
+	}
+
+	// Now, update the IngressController to change to use a NLB and the eipAllocations again.
+	t.Logf("updating ingresscontroller %q to use a NLB and the eipAllocations", ic.Name)
```

Comment:
This is a redundant test. It's already a NLB, so it's not testing anything. My subnet API is dependent on the type, hence why I had a test to switch LB Type, you don't need one.

Reply:
I want to test with valid eipAllocations

Reply:
This test is not switching between lbType